### DISCUSSION TITLE
infinite scroll with take, skip without having page number in the api

### DISCUSSION THREAD

**Discussion by sheilaNasehi** [TanStack/query#2803] [Q&A]
I want to know how can i have infinite scroll with take, skip without having page number in the api.
this is my try but not working:
[https://codesandbox.io/s/strange-architecture-imvhz?file=/src/screens/FollowingsScreen.js](https://codesandbox.io/s/strange-architecture-imvhz?file=/src/screens/FollowingsScreen.js)

**Comment by TkDodo** [comment]
what is `take, skip` and what is not working in that sandbox please?

**Reply by TkDodo** [reply]
what do you mean "not working"?

You don't ned a page number in the api. Whatever you return from `getNextPageParam` will be passed into the `queryFn` as `pageParam`. How you compute the nextPageParam is entirely up to you.
